src/metabase/app_db/connection_pool_setup.clj

```diff
@@ -27,15 +28,42 @@
   []
   (recent-activity?* @latest-activity recent-window-duration))
 
-(defrecord DbActivityTracker []
+(defn- postgres-connection? [^java.sql.Connection connection]
+  (.isWrapperFor connection org.postgresql.jdbc.PgConnection))
+
+;;; I spun these out into separate functions so we can picc up changes to them in the REPL -- Cam
+
+(defn- on-acquire [_connection]
+  (reset! latest-activity (t/offset-date-time)))
+
+(defn- on-check-in [^java.sql.Connection connection]
+  (reset! latest-activity (t/offset-date-time))
+  ;; for Postgres connections, clean up all resources used in the current session. Otherwise we'll just retain a bunch
+  ;; of crap and it can eat up as much as 300MB per connection. See
+  ;; https://www.postgresql.org/docs/current/sql-discard.html for more info about what this does and
+  ;; https://metaboat.slack.com/archives/C05MPF0TM3L/p1748538414384029?thread_ts=1748507464.051999&cid=C05MPF0TM3L for
+  ;; rationale behind why we're doing it -- Cam
+  (when (postgres-connection? connection)
```

Comment:
do we care about running this function on every connection? should this go off the app db type, or perhaps set a value once and then refer to that? or does it not matter?

Reply:
It's fine it's just checking whether the object in question is wraps a specific class of object it's a very inexpensive call under the hood it's just something like 

```clj
(instance? <klass> wrapped-connection)
```